src/tensor.cpp

```diff
@@ -88,28 +88,20 @@ std::optional<miopenTensorLayout_t> GetDefaultLayout(unsigned num_dims)
 }
 
 template <class T>
-bool CheckLengths(const std::vector<T>& lens, T maxval = 0)
+bool CheckLengthsValues(const std::vector<T>& lens)
 {
-    if(lens.empty())
-        return false;
```

Comment:
We should check that the vector is not empty.

Reply:
No, not here, we check non-empty vector later, otherwise we do it multiple times.
This function is dedicated for values sanity check, that's it.